packages/server-wallet/src/protocols/state.ts

```diff
@@ -50,47 +49,54 @@ type SignedByMe = {latestSignedByMe: SignedStateWithHash};
 export type ChannelStateWithMe = ChannelState & SignedByMe;
 export type ChannelStateWithSupported = ChannelState & SignedByMe & WithSupported;
 
-export type Stage = 'Missing' | 'PrefundSetup' | 'PostfundSetup' | 'Running' | 'Final';
 /**
- *
- * @param state: SignedStateWithHash | undefined
- * Returns stage of state.
- *
- * Useful for partitioning the protocol state to decide what action to next take.
+ * The definition of a ChannelStatus is still under debate, but for now
+ * this is what we have settled on. Future changes should include 'funding',
+ * 'defunding', and state related to challenging, probably. See this issue
+ * for more: https://github.com/statechannels/statechannels/issues/2509
  */
-export const stage = (state: State | undefined): Stage =>
-  !state
-    ? 'Missing'
-    : state.isFinal
-    ? 'Final'
-    : state.turnNum <= state.participants.length - 1
-    ? 'PrefundSetup'
-    : state.turnNum <= state.participants.length * 2 - 1
-    ? 'PostfundSetup'
-    : 'Running';
+export const status = (channelState: ChannelState): ChannelStatus => {
```

Comment:
I'm not sure I agree with the change. It is harder for me to understand the implementation of this function, and it appears like it could be incorrect?

Reply:
I think the previous functions were harder to understand and this function more explicitly shows the current logic branch that the previous code was hiding the complexity of — the fact of the matter is that these statuses are somewhat subjective and still not decided upon, which the code also references to the github issue for